File: math/is_prime/Java/is_prime.java
import java.util.*;
public class is_prime {

	public static void main(String[] args) {
		Scanner kb = new Scanner(System.in);
		int n;

		do
		{
			System.out.print("Input number to check if it is prime: ");
			n = kb.nextInt();
			System.out.print("\n");
			if(n<1)
				System.out.println("Please Input a positive number\n");
		}while(n<1);

		kb.close(); //Scanner is no longer needed		
		
		if(n==1){
			System.out.println("1 is not prime");
		}

		for(int i = 2; i<n/2; i++){
			if(n%i==0){
				System.out.println(n + " is not prime");
				return;
			}
		}

		System.out.println(n + " is prime");
	}

}

File: math/is_prime/Lua/isPrime.lua
function isPrime(number)
  --Check if the number is prime
  for i=2,number-1 do
    if number%i == 0 then
      --If remainder number/1 eguals to 0 return not prime
      return "Not prime"
    end
  end
  return "is prime"
end
--call function and print result
print(isPrime(10))